File: Aluguel de Bicicletas em Scala/src/model/DBManager.scala
package model

import java.sql.{Connection, DriverManager}

object DBManager {
  
  def init(){
    createClientTable()
    createBikeTable()
    createRentTable()
    createTransactionTable()
    createBookingTable()
    createStationTable()
  }
  
  def getConnection():Connection = {
    try {
        // db parameters
        val url = "jdbc:sqlite:BikeRent.db"
        // create a connection to the database
        DriverManager.getConnection(url)              
    } catch {
        case e: Exception => e.printStackTrace
        null
    } 
  }
      
  def createClientTable(){
    val sql = "CREATE TABLE IF NOT EXISTS clients (client_id integer PRIMARY KEY," +
             "name text NOT NULL, user text NOT NULL, password text NOT NULL," +
             "cpf varchar(14) NOT NULL, email text NOT NULL, credits double);"
    try{
      val connection = getConnection()
      val stmt = connection.createStatement()
      stmt.executeUpdate(sql);
      stmt.close();
      connection.close()
    }catch{
      case e: Exception => e.printStackTrace
    }
  }
  
  def createBikeTable(){
    val sql = "CREATE TABLE IF NOT EXISTS bikes (bike_id integer PRIMARY KEY," +
             "model text NOT NULL, tax double NOT NULL, state integer NOT NULL," +
             "station_id integer NOT NULL, description text NOT NULL);"
    try{
      val connection = getConnection()
      val stmt = connection.createStatement()
      stmt.executeUpdate(sql);
      stmt.close();
      connection.close()
    }catch{
      case e: Exception => e.printStackTrace
    }
  }
    
  def createRentTable(){
    val sql = "CREATE TABLE IF NOT EXISTS rents (rent_id integer PRIMARY KEY," +
             "rent_date date NOT NULL, client_id integer NOT NULL, bike_id integer NOT NULL," +
             "start time NOT NULL, finish time NOT NULL, value double);"
    try{
      val connection = getConnection()
      val stmt = connection.createStatement()
      stmt.executeUpdate(sql);
      stmt.close();
      connection.close()
    }catch{
      case e: Exception => e.printStackTrace
    }
  }
      
  def createTransactionTable(){
    val sql = "CREATE TABLE IF NOT EXISTS transactions (transaction_id integer PRIMARY KEY," +
             "t_date date NOT NULL, t_type integer NOT NULL, client_id integer NOT NULL," +
             "value double);"
    try{
      val connection = getConnection()
      val stmt = connection.createStatement()
      stmt.executeUpdate(sql);
      stmt.close();
      connection.close()
    }catch{
      case e: Exception => e.printStackTrace
    }
  }
        
  def createBookingTable(){
    val sql = "CREATE TABLE IF NOT EXISTS bookings (booking_id integer PRIMARY KEY," +
             "client_id integer NOT NULL, bike_id integer NOT NULL, booking_date date NOT NULL, start time NOT NULL);"
    try{
      val connection = getConnection()
      val stmt = connection.createStatement()
      stmt.executeUpdate(sql);
      stmt.close();
      connection.close()
    }catch{
      case e: Exception => e.printStackTrace
    }
  }
          
  def createStationTable(){
    val sql = "CREATE TABLE IF NOT EXISTS stations (station_id integer PRIMARY KEY," +
             "name text NOT NULL, adress text NOT NULL);"
    try{
      val connection = getConnection()
      val stmt = connection.createStatement()
      stmt.executeUpdate(sql);
      stmt.close();
      connection.close()
    }catch{
      case e: Exception => e.printStackTrace
    }
  }
}

File: Sistema Bancário Distribuido em Java/BankServerHSBC/src/banco/model/DBManager.java
package banco.model;

import banco.Request;
import java.sql.Connection;
import java.sql.Date;
import java.sql.DriverManager;
import java.sql.Statement;
import java.sql.Time;
import java.util.Calendar;


public class DBManager {
    
    public static Connection getConnection(){
        System.out.println("Conectando com o banco de dados...");
        try {
            // db parameters
            String url = "jdbc:sqlite:Bank.db";
            // create a connection to the database
            return DriverManager.getConnection(url);
            
        } catch(Exception e){
            e.printStackTrace();
            throw new RuntimeException(e);
        }
      }
    
    public static void init(){
        createContasTable();
        createTransacaoTable();
    }
    
    public static void registerTransaction(Request req){
        TransacaoDAO tdao = new TransacaoDAO();
        Transacao t = new Transacao();
        if(req.getBanco_origem() != null){
            t.setBanco_origem(req.getBanco_origem());
        }
        if(req.getBanco_destino()!=null){
            t.setBanco_destino(req.getBanco_destino());
        }
        if(req.getConta_origem()!=null){
            t.setConta_origem(req.getConta_origem());
        }
        if(req.getConta_destino()!=null){
            t.setConta_destino(req.getConta_destino()); 
        }               
        t.setValor(req.getValor());
        Date date = new Date(Calendar.getInstance().getTime().getTime());
        Time horario = new Time(Calendar.getInstance().getTime().getTime());
        t.setData(date);
        t.setHorario(horario);
        switch(req.getOperacao()){
            case "Saque":{
                t.setType(0);
                break;
            }case "Deposito":{
                t.setType(1);
                break;
            }case "Transferencia Env":{
                t.setType(2);            
                break;
            }case "Transferencia Rec":{
                t.setType(3);              
                break;
            }case "TEDCliente":{
                t.setType(4);
                break;
            }case "TEDBanco":{
                t.setType(5);
                break;
            }             
        }
        tdao.inserirTransacao(t);
    }
    
    private static void createContasTable(){
        String sql = "CREATE TABLE IF NOT EXISTS contas (numero integer PRIMARY KEY," +
             "cliente text NOT NULL, senha integer NOT NULL, saldo double NOT NULL," +
             "cpf varchar(14) NOT NULL, data_nascimento data NOT NULL);";
        try{
            Connection conn = getConnection();
            Statement stmt = conn.createStatement();
            stmt.execute(sql);
            stmt.close();
        }catch(Exception e){
            throw new RuntimeException(e);
        }
    }
    
    private static void createTransacaoTable(){
        String sql = "CREATE TABLE IF NOT EXISTS transacoes (id integer PRIMARY KEY," +
                      "type integer NOT NULL, conta_origem integer NOT NULL, "
                      + "conta_destino integer NOT NULL, banco_origem integer NOT NULL, "
                      + "banco_destino integer NOT NULL, data data NOT NULL,"
                      + "horario time NOT NULL, valor double NOT NULL);";
        try{
            Connection conn = getConnection();
            Statement stmt = conn.createStatement();
            stmt.execute(sql);
            stmt.close();
        }catch(Exception e){
            throw new RuntimeException(e);
        }
    }
}
